Experimental

vishy1618 opened the pull request:
I have changed the list_nodes method to have an additional parameter of "ex_node_ids". This will be an array of node ids which will be used to filter the list of nodes returned.

Kami committed:
Move openstack fixtures.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1183573 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Also include tox.in when building a dist package.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1183574 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for 1.1 API and many other improvements for the OpenStack driver.
This patch is part of LIBCLOUD-83. External contributors include Mike Nerone and
Brad Morgan.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1183722 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1183723 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Change version to -beta1.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1184798 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changelog (mostly style fixes.)

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1184804 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for OpenNebula 3.0 API. This patch has been contributed by Hutson
Betts <hut606 at gmail dot com> and it's part of LIBCLOUD-120.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1184967 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix softlayer API url. Their docs are a lie, old URL doesn't work anymore.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186273 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186274 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix softlayer tests.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186280 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a typo in __all__ in the OpenStack compute driver (reported by Brad Morgan).


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186336 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
uh, VERIFY_SSL_CERT needs to default to True.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186461 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Export Terremark provider via get_provider.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186462 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
forgot to update this file.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186505 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update HACKING file and add a new RELEASING file which contains info about 
creating a release.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186534 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify ParamikoSSHClient.connect so it supports authentication using a key file.
This patch has been submitted by Jay Doane <issues dot apache dot org at jay dot
m82 dot com> and it's part of LIBCLOUD-116.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186566 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a test case for ParamikoSSHClient.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1186572 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
User must now explicitly specify a path when LIBCLOUD_DEBUG env. variable is used. Reported by daveb, part of LIBCLOUD-95.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187006 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more attributes to the node.extra dictionary in the EC2 driver. Contributed
by Juan Carlos Moreno part of GITHUB-31.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187584 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a custom NodeSize class with "cpu" and "vcpu" attribute to the OpenNebula
3.0 driver. Contributed by Hutson Betts <hut606 at gmail dot com> part of
LIBCLOUD-119.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187687 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187693 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
OpenStack and Opsource driver style and code cleanup. No functional changes.
Contributed by Hutson Betts <hut606 at gmail dot com> part of LIBCLOUD-123.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187872 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add OpenNebulaNetwork class and ex_list_networks() method to the OpenNebula
driver. Also be consistent and make node states constants uppercase and clean up
the style (double quotes to single quotes, etc.).

Contributed by Hutson Betts <hut606 at gmail dot com> and part of LIBCLOUD-121.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187946 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Refactor response classes and add a new XmlResponse and JsonResponse class.
Modify driver-specific response classes to inherit one of those two classes
where applicable.

Contributed by Caio Romão <caioromao at gmail dot com>, part of LIBCLOUD-97.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187966 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add tests for new response classes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1187970 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Add 24GB node size to GoGrid compute driver.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1188012 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1190054 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove assigned but unused variables.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195016 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove assigned but unused variables.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195017 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new "PollingConnection" class which can work with "async" APIs and use it
in the CloudStackConnection class.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195065 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new DNS driver for Rackspace Cloud DNS (US, UK).


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195069 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove a comment.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195078 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195260 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Wrap ex_create_tags call inside create_node in a try / except block and for now ignore tag
creation errors.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195261 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix public and private ip address handling in the EC2 compute driver. Assign an
empty list to the node object if no public / ip address is found instead of
assigning a list with a "None" element.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195269 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use a node returned by _wait_until_running and change the if statement so the
empty list in node.public_ip doesn't evaluate to True.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195277 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
If LIBCLOUD_DEBUG is specified and paramiko library is available, also set
paramiko library log level to debug.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195280 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Support deploying node using a preinstalled SSH key. Also enable SSH key
deployment in the EC2 driver. This patch has been contributed by Russell Haering
<russellhaering at gmail dot com> and is part of LIBCLOUD-42.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195285 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195294 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add some more explanation about (future) functionality of
upload_object_via_stream.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195771 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure to set have_paramiko to False when debug is on and paramiko is not available. Reported by Juan Carlos Moreno.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195942 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update out of date docstring.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195957 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1195973 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add _upload_data method and missing docstrings.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196289 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add exhaust_iterator function and tests for it.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196310 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8, no functional changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196312 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8, no functional changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196313 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add test for _upload_data.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196320 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8, no functional changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196322 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify _upload_object so it also supports iterators and works with providers which don't support chunked encoding.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196339 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Implement upload_object_via_stream in S3 storage driver. Because S3 doesn't
support chunked transfer encoding whole object is buffered in memory when using
this method.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196341 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add supports_chunked_encoding attribute to CloudFiles class.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196342 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add supports_chunked_encoding attribute to AtmosDriver class.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196344 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add supports_chunked_encoding attribute to GoogleStorageDriver class.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196346 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix typo in S3 tests - should be self.driver.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196347 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8, no functional changes.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196351 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use _get_hash_type function instead of always directly instantiating hashlib.md5
function.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196390 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Default to CHUNK_SIZE if chunk_size is not provided.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196394 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add ex_get_metadata and ex_set_metadata methods to the OpenStack 1.1 compute
driver. Contributed by Mark Everett <mark dot everett at rackspace dot com> and
part of GITHUB-33.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196577 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix the request signature generation in the base EC2 compute driver. If the
endpoint is using a non-standard port, append it to the hostname used to
generate the signature. Patch has been contributed by Simon Delamare <simon dot
delamare at ens-lyon dot fr> and is part of LIBCLOUD-125.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196699 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1196700 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a missing argument to the method call inside
LoadBalancer.attach_compute_node and Driver.balancer_attach_compute_node. This
patch has been contributed by Tim Fletcher <mail at tfletcher dot com> and is
part of LIBCLOUD-126


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197169 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add new "unavailable" state to the BrightboxNodeDriver class. This patch has been
contributed by Tim Fletcher <mail at tfletcher dot com> and is part of
LIBCLOUD-127.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197201 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197202 13f79535-47bb-0310-9956-ffa450edef68

Commit:
- Added ex_start_node method to the ec2 compute driver
- Added ex_stop_node method to the ec2 compute driver
- Added a new _get_state_boolean function to get the state of the node while starting or stopping

Kami committed:
Change base LoadBalancer API to use keyword arguments. Also add tests for
LoadBalancer class methods.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197990 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing destroy() method to the LoadBalancer class.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197991 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8, no functional changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197993 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1197995 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add list_record_types() method to the base DNS API and drivers. Update tests and CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198083 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Bump version, update MANIFEST.in


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198099 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update deprecation warning, old paths will be removed in the next release.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198100 13f79535-47bb-0310-9956-ffa450edef68

Commit:
- Changed the EC2 _get_state_boolean function to use findall instead of iterating.
- Added tests for EC2 ex_start_node, ex_stop_node functionality

Commit:
Merge remote-tracking branch 'upstream/trunk' into trunk

Kami committed:
Increase a PollingConnection timeout in the CloudStack connection and fix the
context dictionary creation in the _async_request method. This patch has been
contributed by "gigimon" and is part of GITHUB-36.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198734 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
"publicipaddress" field in the CloudStack driver are not always available. This
patch has been contributed by "gigimon" and is part of GITHUB-36.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198739 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
"network_id" field in the CloudStack driver create_node is optional. Also fix
network retrieval.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198742 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix private IPs retrieval. Private IPs are not always available.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198744 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing 'Destroyed' state to the CloudStack compute driver. This patch has
been contributed by "gigimon" and is part of GITHUB-36.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198747 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Bump CloudStack PollingConnection timeout to 600 seconds, cuz it takes time for
packets to travel from Europe to Australia.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198748 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198751 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
0.6.1, heck yeah.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1198753 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for Auth 2.0 API (keystone) to the OpenStack Auth connection class.
This patch has been submitted by Brad Morgan <morgabra at cloudkick dot com> and
is part of LIBCLOUD-128.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1199274 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
add 0.6.1 to doap file.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1199719 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add list_locations method to the OpenStack driver and fix some inconsistencies in
the OpenStack driver extension method signatures. This patch has been submitted
by Brad Morgan <morgabra at cloudkick dot com> and is part of LIBCLOUD-129.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1200058 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update Amazon EC2 and S3 driver and add support for a new US West 2 - Oregon
region.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1200328 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in OpenStack connection class. morph_action_hook shouldn't ignore
_force_base url attribute if it's present.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1200980 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Rename public_ip and private_ip on the Node object to public_ips and
private_ips. For backward compatibility still allow users to access those two
attributes the old way.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201114 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure public_ips and private_ips attributes are always a list. Also fix an
inconsistency in IBM SBC driver and make sure those two attributes are a list.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201117 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix an inconsistency in OpSource driver and make sure public_ips is an
empty list ([]), not 'unknown'


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201119 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Increase the default poll interval in the Rackspace driver to 2.5
seconds


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201138 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in Rackspace Cloud DNS driver and make sure to throw an
exception if an unexpected status code is returned. Reported by
"jeblair"


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201141 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Actually use poll_interval and don't ignore it. Also bump default poll timeout
to 200 seconds.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201152 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Change Rackspace Cloud DNS driver poll timeout to 30 seconds.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201153 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Set poll_interval to 0.0 in the tests.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201154 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201155 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Expose CLOUDSTACK provider and allow users to instantiate it. This way it can
also be used with an arbitrary CloudStack provider not just with ninefold.com.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1201975 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Cherry-pick changes from 0.6.x.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1202514 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add new "Cluster Compute Eight Extra Large" size to the Amazon EC2
driver.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1202516 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CC1 instance pricing.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1202520 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Added certain api's to the ec2 driver:
ex_describe_all_addresses
ex_associate_addresses
ex_list_nodes

Also the corresponding tests were added.

Commit:
Merge branch 'trunk' of https://github.com/apache/libcloud into trunk

Commit:
Added fixtures

Kami committed:
Add the following extension methods to the EC2 compute driver:
ex_describe_all_addresses, ex_associate_addresses, ex_start_node,
ex_stop_node.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1202734 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update doap_libcloud.rdf


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1203200 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Add support for the RAX-KSKEY extension, which lets you use API keys, not just a password for the keystone auth service

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1203898 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add ex_member_address argument to the Rackspace LoadBalancer driver
list_balancers method.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1204075 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' of https://github.com/apache/libcloud into trunk

Conflicts:
	libcloud/compute/drivers/ec2.py
	test/compute/test_ec2.py

Commit:
- added a parameter to list_nodes to filter the node output

Kami committed:
Update IBM SBC driver create_node docstring.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1204502 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix the CHANGES file. Those entries should be under in development, not 0.6.2 (failed
to merge it properly the first time).


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1204797 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add NS and MX record types to the Linode DNS driver. Reported by Brandon Helwig.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1205005 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' of https://github.com/apache/libcloud into experimental

Kami committed:
If OpenStack Auth 2.0 API is used, also parse out tenant id and name and save it
on the connection class.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1205685 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix OpenStack Auth 2.0 fixture. Current one probably was from some pre-release
version, because it doesn't match the docs and the actual API response.

Also update all the tests and fix the tenant id parsing.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1205708 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Decrease poll_interval for all the drivers which use PollingConnection class.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1205715 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' into experimental

Commit:
- deleted orig files

Commit:
- url longline repair

Kami committed:
Add ex_node_ids argument to the EC2 driver list_nodes method. This patch has
been contributed by Suvish Vt and it's part of GITHUB-39.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1206263 13f79535-47bb-0310-9956-ffa450edef68

Kami commented:
Commit in [r1206263](http://svn.apache.org/viewvc?rev=1206263&view=rev). Thanks.

Commit:
Merge branch 'trunk' into experimental

Kami committed:
Update OpenStack 1.1 driver to comply with the API specs. Need to make another
call to retrieve node name and ip addresses when creating a node,
because the first call only returns an id an the password. Contributed by Dave
King <dave.king@mailtrust.com>.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1208595 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Update CloudSigma Zurich API endpoint address
2. Add new Las Vegas endpoint and driver
3. Allow user to specify drive type when creating a node


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1209253 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Updated to reflect a patch attributed to LIBCLOUD-121 brining full compute support, PEP 8 compliance, and additional driver tests, for OpenNebula.org versions up to 3.2.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1209752 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Updated CHANGES to reflect changes in r1209752.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1209765 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start working on Python 3 compatibility.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209915 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress on py3k.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209919 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing import.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209920 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
OpenNebula driver tests now pass on py3k.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209922 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use libcloud.py3.{urllib,httplib}.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209923 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Bluebox driver tests now pass.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209925 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use items() instead of iteritems().


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209926 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Values passed to base64.bencode need to be bytes. More progress on porting to
py3k.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209930 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use 'in' instead of has_key().


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209932 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress, map calls needs to be wrapped inside list().


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209933 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
keys() needs to be wrapped in a list().


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209934 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
keys() calls needs to be wrapped inside list().


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209935 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
CloudStack driver tests now pass.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209936 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209937 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209941 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress. All the common + compute tests with the exception if OpenStack
ones now pass.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209943 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
All the common + compute tests now pass with Python 3.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209946 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209948 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More progress. All the Loadbalancer tests now pass.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209949 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
All the DNS tests now pass with Python 3.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209950 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Storage tests now pass with Python 3.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209961 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Export if under different names.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209967 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update examples and demo.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209968 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update setup.py


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209970 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix print in a docstring.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209972 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update py3.py.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209974 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update exception handling.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/branches/py3k@1209975 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Merge py3k branch into trunk.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210013 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES and setup.py.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210025 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pep8.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210026 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Re-organize libcloud.utils modules and create a new libcloud.utils package.
Update affected code.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210045 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Some more Python 3 compatibility changes recommended by 2to3. Most of them are
not needed, but still adding them so it won't cause any surprises in the future.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210046 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove old deprecated paths. Also fix some imports.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210112 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update description in setup.py, add script for migrating from the old style (pre
0.5) paths.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210186 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More Python 3 madness.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210231 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a note about byte arrays.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210235 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Also need to use bytes here.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210244 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Bump version, add Python version classifiers to setup.py.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210248 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More Python 3k stuff.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210262 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Explicitly pass encoding to file.open when using Python 3. Otherwise tests might
fail in some circumstances when locate is not set.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210285 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
trigger a build.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210297 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Trigger a build.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210298 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Trigger a build.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210304 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Fix LIBCLOUD_DEBUG mode

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210658 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
For the OpenStack Auth client, don't morph the action with a base path, we are hitting specific versions ourselves

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210659 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update Rackspace test fixtures to work with the new endpoints URL.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210667 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210672 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Bump version.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1210673 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update doap file.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1212448 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add 'extra' attribute to the LoadBalancer object and retrieve all the virtual IP
addresses in the Rackspace LoadBalancer driver. Contributed by Dave King, part of GITHUB-42.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1212531 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Support for retrieving OpenNebula v3.2 instance types, OpenNebula v3.0 network Public attribute support, and additional code coverage tests.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1212736 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Fixed a few PEP8 compliance issues, and removed unused variables.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1212975 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix typos.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1213009 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add many improvements to the Rackspace LoadBalancer driver. This patch has been
contributed by Dave King and is part of GH-43.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1213364 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add implementation for ex_save_image method to the OpenStack 1.1 driver. This
patch has been contributed by Shawn Smith <shawnpsmith at gmail dot com> and
is part of LIBCLOUD-135.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1213566 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Propagate extra keyword arguments passed to the Rackspace driver
connection class (loadbalancer, storage). This patch has been contributed by
Dave King and is part of GH-41.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1213944 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add Brightbox.net LoadBalancer driver. This patch has been contributed by Tim
Fletcher <mail at tfletcher dot com> and is part of LIBCLOUD-136.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1214063 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add protocol info the Rackspace LoadBalancer object 'extra' dictionary. This patch
has been contributed by Dave King and is part of GH-45.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1214415 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for Amazon EC2 new location - South America (Sao Paulo).


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1214643 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in OpenStack driver when 2.0_apikey or 2.0_password
'auth_version' is used. Reported by Shawn Smith.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1214646 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' into experimental

Kami committed:
Add 'extra' attribute to the Member object and populate it in 
the Rackspace driver. This patch has been contributed by Mark Everett
<mark.everett@rackspace.com> and is part of GH-46.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1221010 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add pricing data for Rackspace 30GB node size.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1221260 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Current OpenNebula OCCI implementation does not support a proper restart method. Rather it suspends and resumes. Therefore, restart_node has been removed from the OpenNebula driver.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1221618 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Adds status to the Member object and conditions an 'enum'
(Rackspace driver). This patch has been contributed by Mark Everett and is part
of GH-48.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1221946 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Add 'request_kwargs' kwarg to the 'get_poll_request_kwargs' method.
This argument contains kwargs which were previously used to initiate the poll
request.

2. Add update_balancer method to the base LoadBalancer class.

3. Add update_balancer and ex_update_balancer_no_poll method to the Rackspace
LoadBalancer driver.

This patch has been contributed by Mark Everett <mark.everett@rackspace.com> and
is part of GH-47.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1222444 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add created and updated attribute to the LoadBalancer extra dictionary in
the Rackspace driver. This patch has been contributed by Mark Everett and is
part of GH-49.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1223041 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add list_record_types method to the base DNSDriver class.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1226328 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a typo in Linode driver. This patch has been contributed by Miguel Jacq
(mig at mig5 dot net) and is part of LIBCLOUD-139.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1227610 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Actually fix the hexlify problem in the Linode driver and add a test case for
it.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1227972 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure that the password we generate in the Linode compute driver 
has at least one uppercase / special character. Should fix 
LIBCLOUD-139.
M    linode.py


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1228414 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Enable ex_delete_image method in the OpenStack 1.1 compute driver. This patch
has been contributed by Shawn Smith <shawnpsmith at gmail dot comA> and is 
part of LIBCLOUD-141.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1228809 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a bunch of extension methods to the Rackspace Loadbalancer driver. This
patch has been submitted bye Dave King <dave dot king at mailtrust dot com>
and it's part of LIBCLOUD-140.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1229390 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add ex_list_protocols_with_default_ports method to the Rackspace loadbalancer
driver. This patch has been contributed by Dave King 
<dave dot king at mailtrust dot com> and it's part of LIBCLOUD-142.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1229432 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Return NodeImage instance in the OpenStack 1.1 node driver ex_save_image method.
This patch has been contributed by Shawn Smith <shawnpsmith at gmail dot com>
and is part of LIBCLOUD-138.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1229803 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing fixture.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1229904 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add UuidMixin class and make Node, NodeImage and NodeSize inherit from it. This
patch has been contributed by Keiko Oda and is part of GH-55.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1237774 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update protocol name mapping in the Rackspace loadbalancer driver. This patch
has been submitted by Bill Woodward <bill dot woodward at rackspace dot com> and
is part of LIBCLOUD-144.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1237930 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Added reboot_node to the OpenNebula v3.2 compute driver.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1238107 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing docstrings to NodeDriver class. patch submitted by Shawn Smith and
is part of GH-57.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239442 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for decompressing responses inside the base Response class. Also add
tests.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239468 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Send Accept-Encoding header.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239469 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Include a --compress flag when using LIBCLOUD_DEBUG.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239767 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a public_ips assignment in the Gande.net compute driver. This patch has been
submitted by Aymeric Barantal <mric at gandi dot net> and is part of LIBCLOUD-147.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239804 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239806 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new module for handling decompression of data.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1239825 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify LoggingConnection to decompress data (if compressed), before logging it.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240037 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix the response classes test.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240062 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Updated the list of node states for OpenNebula drivers. ; LIBCLOUD-148

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240451 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix debug mode so it works with Python 3.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240455 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove extra blank line.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240456 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
bump version.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240457 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Modifed OpenNebula compute fixtures for additional test coverage.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1240458 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add 0.8.0 release.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1242045 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update releasing.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1242943 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Allow user to specify condition and weight when adding a member to the Rackspace
loadbalancer driver. This patch has been submitted by Adam Pickeral and is part
of LIBCLOUD-150.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1242944 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Don't lowercase special headers in the S3 storage driver. Reported by Florent
Cayré, part of LIBCLOUD-149.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1243010 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1244858 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make parsing of the Auth API responses in the OpenStack drivers more flexible
and extensible. This patch is part of LIBCLOUD-151 and has been contributed by
Brad Morgan <morgabra at cloudkick dot com>.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1244862 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Include 'created' and 'updated' attribute in the OpenStack 1.1 driver. This
patch has been submitted by Chris Gilmer <chris dot gilmer at gmail dot com>
and is part of LIBCLOUD-155.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1245193 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add an extension method (ex_balancer_attach_members) for attaching 
 multiple members to a load balancer in the Rackspace driver. This
 patch has been submitted by Adam Pickeral <adam dot pickeral at rackspace dot
 com> and is part of LIBCLOUD-152.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1245771 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Alow users to use a list of tuples for the query string parameters inside
the OpenStack connection classes. This way same the key can be specified
multiple times. This patch has been submitted by Dave King and is part of
LIBCLOUD-153.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1245823 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix curl command which is logged when using debug mode to reflect the actual
protocol (http, https) which is used inside the url.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1245889 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add __repr__ for all the common types. This addresses LIBCLOUD-156.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1290936 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify OpenStack driver to also handle JSON error responses without the
'message' attribute.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1290945 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Allow user to pass 'max_tries' keyword argument to deploy_node method.
2. Include original exception error message when re-throwing an exception
inside _run_deployment_script method.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1291117 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start working on libvirt driver.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1291129 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use .info() to retrieve node state and include some other attributes in extra
dictionary.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1291130 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add support for ElasticHosts new United States (Los Angeles) and Canada
(Toronto) locations. This patch has been submitted by Jaime Irurzu and is part
of GH-53.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1291132 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug with handling decompressed response in the Linode drivers introduced
in 0.8.0. This patch has been submitted by Ben Agricola <ben dot agricola at
gmail dot com> and is part of LIBCLOUD-158.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1293688 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add serverId attribute to the NodeImage object extra dictionary in the
OpenStack driver. This patch has been contributed by Mark Everett and is part of
GITHUB-61.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1296520 13f79535-47bb-0310-9956-ffa450edef68

hutson committed:
Updated some source code to comply with PEP8.

git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1297286 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Include 'minRam' and 'minDisk' attribute in the OpenStack 1.1 driver 
Node extra dictionary. This patch has been contributed by Chris Gilmer and is
part of LIBCLOUD-163.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1298491 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add new Amazon instance type - m1.medium.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1298501 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Allow user to re-use auth tokens and pass 'ex_force_auth_token' keyword
argument to the OpenStack driver constructor. This patch has been contributed by
Dave King and is part of LIBCLOUD-164.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1300356 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' into experimental

Commit:
deleted all

Commit:
Revert "deleted all"

This reverts commit e2aaa88a127eaef054ee8595958a26f041769f97.

Kami committed:
Add ex_creaate_balancer method to the Rackspace driver and allow user to
pass 'vip' argument to it. THis patch has been contributed by Adam Pickeral and
is part of LIBCLOUD-166.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1301126 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new mixin class and make all the OpenStack based drivers inherit from it.
This way all the OpenStack based drivers accept the same extenstion keyword
arguments. This patch has been contributed by Dave King and is part of
LIBCLOUD-167.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1301741 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Change the code so it works with all the supported versions.


git-svn-id: https://svn.apache.org/repos/asf/libcloud/trunk@1301756 13f79535-47bb-0310-9956-ffa450edef68

Commit:
Merge branch 'trunk' into experimental

Commit:
Got basic list_balancers call working

Commit:
- elb is now almost base feature complete

Commit:
- added some function documentation

Kami commented:
(just doing some pull request cleanup)

@vishy1618 - Those changes are not relevant anymore, can you please close this pull request?

Thanks!

vishy1618 commented:
Sorry for not being too involved on this one, I'll close it now.

On Sat, Oct 26, 2013 at 7:00 PM, Tomaz Muraus notifications@github.comwrote:

> (just doing some pull request cleanup)
> 
> @vishy1618 https://github.com/vishy1618 - Those changes are not
> relevant anymore, can you please close this pull request?
> 
> Thanks!
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/apache/libcloud/pull/39#issuecomment-27146252
> .

## 

Suvish V.T.
